solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+  name: 'cc-${location}-spokesnonprod'
+  parent: networkManager
+  dependsOn: [
+    networkManager::networkGroupNonProd::staticMembersSpokeOne
+    networkManager::networkGroupNonProd::staticMembersSpokeTwo
+  ]
+  properties: {
+    description: 'Non-prod poke-to-spoke connectivity configuration'
+    displayName: 'Non-prod Spoke-to-Spoke Connectivity'
+    appliesToGroups: [
+      {
+        networkGroupId: networkManager::networkGroupNonProd.id
+        isGlobal: 'False'
+        useHubGateway: string(deployVpnGateway)
+        groupConnectivity: 'DirectlyConnected'
+      }
+    ]
+    connectivityTopology: 'HubAndSpoke'
+    deleteExistingPeering: 'True'
+    hubs: [ 
+      {
+        resourceId: hubVnetId
+        resourceType: 'Microsoft.Network/virtualNetworks'
+      } 
+    ]
+    isGlobal: 'False'
+  }
+}
+
+@description('This connectivity configuration defines the connectivity between the spokes. Only deployed if requested.')
+resource connectivityConfigurationProd 'Microsoft.Network/networkManagers/connectivityConfigurations@2022-05-01' = {
+  name: 'cc-${location}-spokesprod'
+  parent: networkManager
+  dependsOn: [
+    networkManager::networkGroupProd::staticMembersSpokeOne
+    networkManager::networkGroupProd::staticMembersSpokeTwo
+  ]
+  properties: {
+    description: 'Prod spoke-to-spoke connectivity configuration (through hub)'
+    displayName: 'Prod Spoke-to-Spoke Connectivity'
+    appliesToGroups: [
+      {
+        networkGroupId: networkManager::networkGroupProd.id
+        isGlobal: 'False'
+        useHubGateway: string(deployVpnGateway)
+        groupConnectivity: 'None'
+      }
+    ]
+    connectivityTopology: 'HubAndSpoke'
+    deleteExistingPeering: 'True'
+    hubs: [ 
+      {
+        resourceId: hubVnetId
+        resourceType: 'Microsoft.Network/virtualNetworks'
+      } 
+    ]
+    isGlobal: 'False'
+  }
+}
+
+@description('This user assigned identity is used by the Deployment Script resource to interact with Azure resources.')
+resource userAssignedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2022-01-31-preview' = {
+  name: 'uai-${location}'
+  location: location
+}
+
+@description('This role assignment grants the user assignmed identity the Contributor role on the resource group.')
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(resourceGroup().id, userAssignedIdentity.name)
+  properties: {
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c') // Contributor: b24988ac-6180-42a0-ab88-20f7382dd24c
+    principalId: userAssignedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+@description('This is the securityadmin configuration assigned to the AVNM')
+resource securityConfig 'Microsoft.Network/networkManagers/securityAdminConfigurations@2022-05-01' = {
+  name: 'sg-${location}'
+  parent: networkManager
```

Comment:
Another child that isn't declared in the parent resource -- continue to be curious on these.  I'll stop mentioning them, but there are more in here